### PR TITLE
Apply rust-logo class only on default logo

### LINKED ISSUE #91653: Adaptive SVG logos look terrible on dark themes
In RustCrypto crates we use [SVG logo](https://raw.githubusercontent.com/RustCrypto/media/master/logo.svg) which changes style depending on preferred theme using `@media(prefers-color-scheme:dark) { .. }` CSS rules. You can see the [`sha2` docs](https://docs.rs/sha2/latest/sha2/) for an example which uses it.

But the problem is how the logo gets rendered on dark themes:
ayu
dark

The reason for such horrible result is the white shadow filter applied to the logo. I understand why it was added, but I think it's a wrong approach, especially for projects which aim to properly support dark themes in their media files.

### PR DESCRIPTION
**Pull request by jsha**
Fixes #91653.

image

Demo: https://rustdoc.crud.net/jsha/hashes/sha2/

r? @GuillaumeGomez

### CODE REVIEW

**Comment by jhpratt** [comment]
I feel like having an environment variable would make most sense. Either opt-in or opt-out; I'd personally prefer the latter. That way we don't keep switching back and forth.

File: src/librustdoc/html/templates/page.html [hunk 1]
```diff
@@ -75,15 +75,15 @@
...
+            {%- else -%}
             <div class='logo-container rust-logo'> {#- -#}
-                <img src='
-                    {%- if layout.logo -%}
-                    {{layout.logo}}
-                    {%- else -%}
-                    {{static_root_path | safe}}rust-logo{{page.resource_suffix}}.png
-                    {%- endif -%}
-                    ' alt='logo'> {#- -#}
+                <img src='{{static_root_path | safe}}rust-logo{{page.resource_suffix}}.png'> {#- -#}
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/templates/page.html:84]: You forgot the `alt='logo'`. ;)

File: src/librustdoc/html/templates/page.html [hunk 2]
```diff
@@ -74,30 +74,30 @@
...
+            {%- endif -%}
         </a> {#- -#}
         {{- sidebar | safe -}}
     </nav> {#- -#}
     <main> {#- -#}
         <div class="width-limiter"> {#- -#}
             <div class="sub-container"> {#- -#}
-                <a class="sub-logo-container rust-logo" href='{{page.root_path | safe}}{{krate_with_trailing_slash | safe}}index.html'> {#- -#}
-                    <img src='
+                <a class="sub-logo-container rust-logo" href="{{page.root_path | safe}}{{krate_with_trailing_slash | safe}}index.html"> {#- -#}
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/templates/page.html:93]: Just realized it: you need the condition here as well. This part is to display the logo in the sidebar on mobile.

**Comment by jsha** [comment]
I fixed the missing `alt`. I also took the opportunity to replace some inconsistent `'` with `"` (which we use everywhere else).

@jhpratt said:

> I feel like having an environment variable would make most sense. Either opt-in or opt-out; I'd personally prefer the latter. That way we don't keep switching back and forth.

I don't understand this comment.

**Comment by GuillaumeGomez** [comment]
With my second commit, I realized that we didn't have a test to ensure that the image in the `sub-logo` was the right one. Can you add one or open an issue please? If you prefer to not do it in this PR, r=me (but still open the issue ;) ).

**Review by GuillaumeGomez — APPROVED** [review]

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Awesome, thanks! r=me once CI pass

**Comment by jhpratt** [comment]
> I don't understand this comment.

If we used an environment variable to control whether the outline is present, then it would be fully under control of the crate. If I remember correctly, whether the outline applies to every logo has gone back and forth a few times. Personally I would like to have it for my crates, but others (such as you) don't. So why not make it configurable? 🤷‍♂️

**Comment by GuillaumeGomez** [comment]
You can instead use `--html-in-header` and add the style you want.

**Comment by jhpratt** [comment]
Certainly. But for common things, which I imagine this is, configuration makes sense imo.

**Comment by jsha** [comment]
> whether the outline applies to every logo has gone back and forth a few times

I don't know the full history here, but at least the most recent change here was an unintentional bug, not a design choice, which is what I'm trying to fix. I'm not aware of a time when outlining was intentionally applied to non-default logos.

**Comment by GuillaumeGomez** [comment]
@jhpratt I don't think having such things handled by environment variables will be any better. Also, it was overwhelmingly in favor of not changing the display of the logos provided by users. Use the existing options.

@jsha I can't find the link but in short it was something I think to have a "border" applied on all logos to make them look "good" whatever the theme.

**Pull request merged**